### PR TITLE
Hand in hand form

### PR DESCRIPTION
**Pull request by rickifox**

### CODE REVIEW

**Review by KevinRoig — DISMISSED** [review]
ok

File: src/Service/ImageValidatorNAdder.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace App\Service;
+
+use App\Entity\Image;
+use App\Entity\Article;
+use App\Repository\ImageRepository;
+
+class ImageValidatorNAdder
+{
+    public function addImageIf(string $savedImageId, Article $article, ImageRepository $imageRepository): Article
```
Comment (RomainClair) [inline src/Service/ImageValidatorNAdder.php]: You could inject Services in Services using dependency injection in the service constructor

File: src/Controller/ImageFormController.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        $imageForm->handleRequest($request);
+        if ($imageForm->isSubmitted() && $imageForm->isValid()) {
+            $imageIds = $imageForm->get('otherImages')->getData();
+            $entityManager->persist($image);
+            $entityManager->flush();
+            $imageIds = $idsListGenerator->setImageIdsList($imageIds, $image);
+            $imageIdsArr = explode(', ', $imageIds);
+            foreach ($imageIdsArr as $imageId) {
+                $images[] = $imageRepository->find($imageId);
+            }
```
Comment (RomainClair) [inline src/Controller/ImageFormController.php]: The setImageIdsList() could do this too and return the Image array directly
in a addAndGetImages() method for example

File: src/Controller/ArticleFormController.php [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        $imageForm->handleRequest($request);
+        if ($imageForm->isSubmitted() && $imageForm->isValid()) {
+            $imageIds = $imageForm->get('otherImages')->getData();
+            $entityManager->persist($image);
+            $entityManager->flush();
+            $imageIds = $idsListGenerator->setImageIdsList($imageIds, $image);
+            $imageIdsArr = explode(', ', $imageIds);
+            foreach ($imageIdsArr as $imageId) {
+                $images[] = $imageRepository->find($imageId);
+            }
```
Comment (RomainClair) [inline src/Controller/ArticleFormController.php]: The setImageIdsList() could do this too and return the Image array directly
in a addAndGetImages() method for example

File: src/Controller/ArticleFormController.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        EntityManagerInterface $entityManager,
+        ImageRepository $imageRepository,
+        IdsListGenerator $idsListGenerator,
+        ImageValidatorNAdder $imageValidatorNAdder
+    ): Response {
+        $image = new Image();
+        $imageForm = $this->createForm(ImageFormType::class, $image);
+        $article = new Article();
+        $articleForm = $this->createForm(ArticleFormType::class, $article);
+        $imageIds = '';
```
Comment (RomainClair) [inline src/Controller/ArticleFormController.php:33]: This is useless (imageIds are modified after without being read)

File: src/Controller/ArticleFormController.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+                $imagesArray = explode(', ', $imageIds);
+                foreach ($imagesArray as $savedImageId) {
+                    $article = $imageValidatorNAdder->addImageIf($savedImageId, $article, $imageRepository);
+                }
+                $imageIdsArr = explode(', ', $imageIds);
+                foreach ($imageIdsArr as $imageId) {
+                    $images[] = $imageRepository->find($imageId);
+                }
+                $entityManager->persist($article);
+                $entityManager->flush();
```
Comment (RomainClair) [inline src/Controller/ArticleFormController.php]: All this should be the Service method

**Review by RomainClair — CHANGES_REQUESTED** [review]
Your IdsListGenerator and ImageValidatorNAdder should probably be only one service because they take part of the same functionnality

The ArticleFormController should probably be just an ArticleController and same for the ImageFormController (or maybe the only controller in ImageFormController should be in the ArticleController

Remeber : Don't Repeat Yourself !

File: src/Controller/ArticleFormController.php (hunk 5 above)
Comment (rickifox) [reply src/Controller/ArticleFormController.php]: Done

File: src/Service/ImageValidatorNAdder.php (hunk 1 above)
Comment (rickifox) [reply src/Service/ImageValidatorNAdder.php]: Kesako?

**Review by KevinRoig — APPROVED** [review]
ok

File: src/Controller/ImageFormController.php (hunk 2 above)
Comment (rickifox) [reply src/Controller/ImageFormController.php]: Done

File: src/Controller/ArticleFormController.php (hunk 4 above)
Comment (rickifox) [reply src/Controller/ArticleFormController.php:33]: Tell that to stan

File: src/Controller/ArticleFormController.php (hunk 3 above)
Comment (rickifox) [reply src/Controller/ArticleFormController.php]: Done

**Review by RomainClair — APPROVED** [review]

**Pull request merged**